plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1155,50 +1159,90 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
                     throw new TrinoException(NOT_SUPPORTED, "Updating the " + trinoPropertyName + " property is not supported");
             }
         }
 
         try {
-            updateProperties.commit();
+            updateProperties.ifPresent(PendingUpdate::commit);
+            if (properties.containsKey(PARTITIONING_PROPERTY)) {
+                updatePartitioning(icebergTable, transaction, (List<String>) properties.get(PARTITIONING_PROPERTY).orElseThrow());
+            }
             transaction.commitTransaction();
         }
         catch (RuntimeException e) {
             throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit new table properties", e);
         }
     }
 
-    private static void updateProperty(UpdateProperties updateProperties, String icebergPropertyName, Optional<Object> value, Function<Object, String> toIcebergString)
+    private static UpdateProperties updateProperty(UpdateProperties updateProperties, String icebergPropertyName, Optional<Object> value, Function<Object, String> toIcebergString)
     {
         if (value.isPresent()) {
             updateProperties.set(icebergPropertyName, toIcebergString.apply(value.get()));
         }
         else {
             updateProperties.remove(icebergPropertyName);
         }
+
+        return updateProperties;
+    }
+
+    private static void updatePartitioning(Table icebergTable, Transaction transaction, List<String> partitionColumns)
+    {
+        UpdatePartitionSpec updatePartitionSpec = transaction.updateSpec();
+        Set<PartitionField> existingPartitionFields = icebergTable.spec().fields().stream().collect(toImmutableSet());
+        Schema schema = icebergTable.schema();
+        if (partitionColumns.isEmpty()) {
+            existingPartitionFields.stream()
+                    .map(partitionField -> toIcebergTerm(schema, partitionField))
+                    .forEach(updatePartitionSpec::removeField);
+        }
+        else {
+            List<PartitionField> partitionFields = parsePartitionFields(schema, partitionColumns).fields();
+            existingPartitionFields.stream()
+                    .filter(partitionField -> !partitionFields.contains(partitionField))
+                    .map(partitionField -> toIcebergTerm(schema, partitionField))
+                    .forEach(updatePartitionSpec::removeField);
+            partitionFields.stream()
+                    .filter(partitionField -> !existingPartitionFields.contains(partitionField))
+                    .map(partitionField -> toIcebergTerm(schema, partitionField))
+                    .forEach(updatePartitionSpec::addField);
+        }
+
+        updatePartitionSpec.commit();
+    }
+
+    private static Term toIcebergTerm(Schema schema, PartitionField partitionField)
+    {
+        return Expressions.transform(schema.findColumnName(partitionField.sourceId()), partitionField.transform());
```

Comment:
It's kind of weird that `PartitionSpec` cannot be used directly in `UpdatePartitionSpec` and we need to go thru expressions like this.

Do you think it's something we could change in Iceberg API?
cc @rdblue @danielcweeks 

(that would be a follow-up for this PR, no change requested here)